packages/calypso-e2e/src/lib/pages/media-page.ts

```diff
@@ -0,0 +1,152 @@
+import { waitForElementEnabled } from '../../element-helper';
+
+/**
+ * Type dependencies
+ */
+import { Page } from 'playwright';
+
+const selectors = {
+	// Navigation tabs
+	navTabs: '.section-nav-tabs',
+	navTabsDropdownOptions: '.select-dropdown__option',
+
+	// Gallery view
+	gallery: '.media-library__content',
+	items: '.media-library__list-item',
+	placeholder: '.is-placeholder',
+	editButton: 'button[data-e2e-button="edit"]',
+
+	// Modal view
+	mediaModal: '.editor-media-modal__content',
+	mediaModalEditButton:
+		'.editor-media-modal-detail__edition-bar .editor-media-modal-detail__edit:visible',
+	mediaModalPreview: '.editor-media-modal-detail__preview',
+
+	// Editor view
+	imageEditorCanvas: '.image-editor__canvas-container',
+	imageEditorToolbarButton: '.image-editor__toolbar-button',
+	imageEditorResetButton: 'button[data-e2e-button="reset"]',
+	imageEditorCancelButton: 'button[data-e2e-button="cancel"]',
+};
+
+/**
+ * Represents an instance of the WPCOM Media library page.
+ *
+ * @augments {BaseContainer}
+ */
+export class MediaPage extends BaseContainer {
+	/**
+	 * Constructs an instance of the MediaPage object.
+	 *
+	 * @param {Page} page Underlying page on which interactions take place.
+	 */
+	constructor( page: Page ) {
+		super( page, selectors.gallery );
+	}
+
+	/**
+	 * Post-initialization steps.
+	 *
+	 * @returns {Promise<void>} No return value.
+	 */
+	async _postInit(): Promise< void > {
+		await this.page.waitForLoadState( 'domcontentloaded' );
+	}
+
+	/**
+	 * Given a 1-indexed number `n`, click and select the nth item in the media gallery.
+	 *
+	 * Note that if the media gallery has been filtered (eg. Images only), this method
+	 * will select the `nth` item in the filtered gallery as shown.
+	 *
+	 * @param {number} index 1-indexed value denoting the nth media gallery item to be selected.
+	 * @returns {Promise<void>} No return value.
+	 * @throws {Error} If requested item could not be located in the gallery, or if the click action
+	 * failed to select the gallery item.
+	 */
+	async selectItem( index: number ): Promise< void > {
+		// Playwright is able to select the nth matching item given a selector.
+		// See https://playwright.dev/docs/selectors#pick-n-th-match-from-the-query-result.
+		const element = await this.page.waitForSelector(
+			`:nth-match(${ selectors.items }, ${ index })`
+		);
+
+		// Wait for the target element to be stable before checking for the `is-selected` class.
+		// Otherwise, Playwright executes too fast and the method throws despite successfully selecting.
+		await element.waitForElementState( 'visible' );
+		await element.click();
+		await element.waitForElementState( 'stable' );
+		const classAttributes = ( await element.getAttribute( 'class' ) ) as string;
+		const isSelected = classAttributes.includes( 'is-selected' );
+		if ( ! isSelected ) {
+			throw new Error( `Failed to select requested item number ${ index }` );
+		}
```

Comment:
We should be able to skip the middle checks here if we wait for the `is-selected` class after the click:
```suggestion
		// Playwright is able to select the nth matching item given a selector.
		// See https://playwright.dev/docs/selectors#pick-n-th-match-from-the-query-result.
		const elementHandle = await this.page.click( `:nth-match(${ selectors.items }, ${ index })` );
		await this.page.waitForFunction(
			( element ) => element.classList.contains( 'is-selected' ),
			elementHandle
		);
```

Reply:
This change results in the following:

```
page.waitForFunction: Evaluation failed: TypeError: Cannot read property 'classList' of undefined
            at eval (eval at predicate (eval at evaluate (:303:29)), <anonymous>:1:22)
            at predicate (eval at evaluate (:303:29), <anonymous>:5:30)
            at eval (eval at evaluate (:303:29), <anonymous>:18:78)
            at onRaf (<anonymous>:1667:37)
            at <anonymous>:1677:13
            at Object.run (<anonymous>:1735:28)
            at eval (eval at evaluate (:303:29), <anonymous>:1:14)
            at UtilityScript.evaluate (<anonymous>:305:22)
            at UtilityScript.<anonymous> (<anonymous>:1:44)
```

VSCode suggests that `element` is of `void` type, so that probably has something to do with it.

Reply:
Yeah, so the proposed change will not work because `page.click` returns a `Promise<void>`, not the ElementHandle.

Therefore, it would still be better to keep Line 74 from the original change and modify the suggested change to reference the ElementHandle returned in that line.

Reply:
Ah, nice catch! 👍